website/docs/opengl/c1/2-hello-quad.md

```diff
@@ -0,0 +1,524 @@
+
+`positionLoc` simply refers to the `aPosition` attribute we defined in our vertex shader earlier. Since we explicitly stated we wanted it at position 0, we must also use that here. If you decided against explicitly giving a location in your shader, you can replace the `0` with `_gl.GetAttribLocation("aPosition");` (just remember to remove the `const` too...)
+
+Before we can use the attribute, we must first enable it. Then, we tell it what part of the data to look at.
+
+First, we tell it the size of the attribute's data type. Since we're using a `vec3`, we tell it that the size is 3. Next up, we tell it that we're using floats. This is the most common pointer type used. The `false` tells it we don't want to normalize the values. If we did normalize them, OpenGL would convert them to between 0 and 1. Finally, we have the **stride** and **offset**.
+
+The stride tells OpenGL the size (in bytes) of a *single* vertex. The offset tells OpenGL the offset *within* the stride of the attribute.
+
+This diagram gives a visual explanation of what stride and offset do (credit to LearnOpenGL):
+
+![stride and offset](../../../images/opengl/chapter1/vertex_attribute_pointer.png)
+
+In our example, the only things we define per vertex is the position of the vertex itself, which is 3 values per vertex. Therefore, our stride is just `3 * sizeof(float)` (remember, stride is in **bytes**, so we must multiply by the size of the float primitive). Since we are only defining one attribute, we don't need to have any offset. Therefore, we can just use `0`. OpenGL expects a `void` pointer, so we must cast it to `void*`.
+
+### Cleaning up
+We're nearly there. The last thing we should do is to unbind the various buffers. While you don't *need* to do this, it will help to reduce the risk of changing a value for the wrong buffer.
+
+Add the following code to your `OnLoad` method:
+
+```
+_gl.BindVertexArray(0);
+_gl.BindBuffer(BufferTargetARB.ArrayBuffer, 0);
+_gl.BindBuffer(BufferTargetARB.ElementArrayBuffer, 0);
+```
+
+Doing this means we've "un-bound" everything, so calling something like `BufferData` won't affect the buffers we've just created.
+
+<?# Warning "You **MUST** unbind the vertex array first, before unbinding the other buffers. If you forget to do it in this order, the buffer will be unbound from the vertex array, meaning you'll see incorrect results when you render the object." /?>
+
+If you want to see the resulting code so far, you can see it [here](../sources/1.2.7-finished-setup.html).
+
+## Drawing to the screen
+It's finally time to draw to the screen! That was a lot of setup work we just did there. Fortunately, that was the hard part. Drawing our result to the screen is now very easy.
+
+Add the following to your `OnRender` method:
+
+```cs
+_gl.BindVertexArray(_vao);
+_gl.UseProgram(_program);
+_gl.DrawElements(PrimitiveType.Triangles, 6, DrawElementsType.UnsignedInt, null);
```

Comment:
```suggestion
_gl.DrawElements(PrimitiveType.Triangles, 6, DrawElementsType.UnsignedInt, (void*) 0);
```

I think it's better if we avoid `null` on these numeric values. It is confusing enough that numeric values have to be passed as pointers, even though they don't point to anything! I hope for 3.0 we can turn all of these into `uint`s or something ;_;

Reply:
Yeah it would definitely be good to have it as a uint, or at least an overload for it. I often (including myself when I switched to silk) see people using `0` by itself and then being confused as to why it didn't work. While I personally use `null`, for the purposes of the tutorial I agree that `(void*) 0` is better. I'll leave this unresolved for now so I remember to update all the files containing this code (and not just the tutorial) when I make these changes tomorrow.